### PR TITLE
Align and simplify the stable training scripts

### PR DESCRIPTION
**Pull request by qgallouedec**
The recommended way to instantiate a trainer is to pass a model id string and let the trainer call `from_pretrained` itself (via `model_init_kwargs`). The CLI scripts under `trl/scripts/` were not following this — most loaded the model manually, and they had drifted apart in how they built `model_init_kwargs`.

This PR moves `sft`, `dpo`, `grpo`, `rloo`, `reward` to the same pattern:

```python
training_args.model_init_kwargs = dict(revision=..., trust_remote_code=..., attn_implementation=..., dtype=...)
quantization_config = get_quantization_config(model_args)
if quantization_config is not None:
    training_args.model_init_kwargs["device_map"] = get_kbit_device_map()
    training_args.model_init_kwargs["quantization_config"] = quantization_config
...
trainer = XxxTrainer(model=model_args.model_name_or_path, ...)
```

Along the way:

- **sft**: drops the manual `AutoModelForImageTextToText` branch (the trainer resolves it from `config.architectures[0]`).
- **dpo**: drops a redundant post-train `trainer.evaluate()` block; `ignore_bias_buffers` now operates on `trainer.model`.
- **grpo**: drops a redundant manual `getattr(torch, dtype)` and forwards `trust_remote_code`.
- **rloo** / **reward**: add the missing `model_init_kwargs` block — previously `--dtype` / `--load_in_4bit` / `--attn_implementation` were silently ignored.
- **examples/scripts/sft_nemotron_3.py**: same simplification (direct replica of the old SFT pattern).

`grpo.py` and `rloo.py` are now byte-identical except for class names. Other examples (`cpo`, `orpo`, `bco`, `kto`, `nash_md`, `xpo`, `online_dpo`, `sft_gpt_oss`, `sft_video_llm`, `sft_tiny_aya_tool_calling`) are left alone — each has a twist (multi-model load, custom processor, custom quantization) that doesn't fit cleanly.

---

> [!NOTE]
> **Medium Risk**
> Moderate risk because multiple CLI training entrypoints change how models are instantiated (and how dtype/quantization/trust_remote_code are applied), which can affect runtime behavior across common workflows.
> 
> **Overview**
> **Unifies model instantiation across the “stable” training CLIs.** The `sft`, `dpo`, `grpo`, `rloo`, `reward`, and `examples/scripts/sft_nemotron_3.py` scripts now set `training_args.model_init_kwargs` (revision/`trust_remote_code`/attention impl/dtype, plus k-bit `device_map` + `quantization_config` when enabled) and pass `model=model_args.model_name_or_path` into the trainer, rather than pre-loading the model themselves.
> 
> This removes divergent/duplicated logic (notably the manual `AutoModelForImageTextToText` branch in `sft.py`), ensures `reward.py`/`rloo.py` actually honor dtype/quantization-related flags, and simplifies `dpo.py` by dropping an extra post-train eval block and applying `ignore_bias_buffers` on `trainer.model`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 0441d1eaa810e8b031062a4d6499ab4bd4f67d1e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by kashif — APPROVED** [review]

**Review by albertvillanova — APPROVED** [review]
Thanks.
